include/nbl/builtin/hlsl/ieee754.hlsl

```diff
@@ -130,6 +131,27 @@ NBL_CONSTEXPR_INLINE_FUNC typename unsigned_integer_of_size<sizeof(T)>::type ext
 	return ieee754::impl::bitCastToUintType(x) & traits<AsFloat>::signMask;
 }
 
+template <typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<FloatingPoint>)
+NBL_CONSTEXPR_INLINE_FUNC FloatingPoint copySign(FloatingPoint to, FloatingPoint from)
+{
+	using AsUint = typename unsigned_integer_of_size<sizeof(FloatingPoint)>::type;
+
+	const AsUint toAsUint = ieee754::impl::bitCastToUintType(to) & (~ieee754::traits<FloatingPoint>::signMask);
+	const AsUint fromAsUint = ieee754::impl::bitCastToUintType(from);
+
+	return bit_cast<FloatingPoint>(toAsUint | extractSignPreserveBitPattern(from));
+}
+
+template <typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<FloatingPoint>)
+NBL_CONSTEXPR_INLINE_FUNC FloatingPoint flipSign(FloatingPoint val)
+{
+	using AsFloat = typename float_of_size<sizeof(FloatingPoint)>::type;
+	using AsUint = typename unsigned_integer_of_size<sizeof(FloatingPoint)>::type;
+	const AsUint asUint = ieee754::impl::bitCastToUintType(val);
+
+	return bit_cast<FloatingPoint>(asUint ^ ieee754::traits<AsFloat>::signMask);
```

Comment:
reply to https://github.com/Devsh-Graphics-Programming/Nabla/pull/803#discussion_r1920258107

yes I want a bool saying whether I want to flip or not, that way the function is the mos tuseful